src/Microsoft.ComponentDetection.Contracts/IComponentRecorder.cs

```diff
@@ -45,6 +45,9 @@ public interface ISingleFileComponentRecorder
 
     DetectedComponent GetComponent(string componentId);
 
+    /// <summary>
+    /// Any file added here will be reported as a location on ALL components found in current graph.
+    /// </summary>
     void AddAdditionalRelatedFile(string relatedFilePath);
```

Comment:
In our wrapper we are using this method, but this inadvertently adds it to ALL components found in that file. There currently isn't a way to customize additional locations per component, as we are clearing everything in DefaulGraphTranslationService.cs